File: solutions/python/scrabble-score/1/scrabble_score.py
def score(word):
    letters = {
        'D': 2, 'G': 2,
        'B': 3, 'C': 3, 'M': 3, 'P': 3,
        'F': 4, 'H': 4, 'V': 4, 'W': 4, 'Y': 4,
        'K': 5,
        'J': 8, 'X': 8,
        'Q': 10, 'Z': 10,
    }
    word = word.upper() if word else ""
    result = 0
    for letter in word:
        if not letter.isalpha():
            continue
        result += letters.get(letter, 1)
    return result

File: solutions/go/scrabble-score/1/scrabble_score.go
package scrabble

import "strings"

func Score(word string) int {
	letters := map[rune]int{
		'D': 2, 'G': 2,
		'B': 3, 'C': 3, 'M': 3, 'P': 3,
		'F': 4, 'H': 4, 'V': 4, 'W': 4, 'Y': 4,
		'K': 5,
		'J': 8, 'X': 8,
		'Q': 10, 'Z': 10,
	}
	result := 0
	for _, letter := range strings.ToUpper(word) {
		if letter < 'A' || letter > 'Z' {
			continue
		}
		value, found := letters[letter]
		if !found {
			value = 1
		}
		result += value
	}
	return result
}
